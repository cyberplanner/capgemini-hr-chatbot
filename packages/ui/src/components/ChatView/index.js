import React from "react";
import styled from "styled-components";

const Chat = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  max-height: 100%;
  overflow-y: scroll;
`;

const Message = styled.p`
  background: ${props =>
    props.user.toLowerCase() === "bot" ? "#eceff1" : "#3a96dd"};
  color: ${props => (props.user.toLowerCase() === "bot" ? "#000" : "#fff")};
  align-self: ${props =>
    props.user.toLowerCase() === "bot" ? "flex-start" : "flex-end"};
  width: 80%;
  max-width: 300px;
  border-radius: 15px;
  padding: 10px 20px;
  box-sizing: border-box;
  text-align: left;
`;

export default class ChatView extends React.Component {
  render() {
    let { messages = [] } = this.props;
    return (
      <Chat>
        {messages.map((message, index) => {
          if (!message) {
            return <div />;
          }
          return (
            <Message key={index} user={message.user}>
              {message.text}
            </Message>
          );
        })}
      </Chat>
    );
  }
}
