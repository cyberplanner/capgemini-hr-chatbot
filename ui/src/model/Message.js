export default class Message {
  _user;
  _text;

  constructor(user, text) {
    this._user = user;
    this._text = text;
  }

  get user() {
    return this._user;
  }

  get text() {
    return this._text;
  }
}
