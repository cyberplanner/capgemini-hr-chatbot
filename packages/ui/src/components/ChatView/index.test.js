import React from "react";
import ChatView from "./index";
import Message from "../../model/Message";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const tree = renderer.create(<ChatView />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing with an empty array of messages", () => {
  let messages = [];

  const tree = renderer.create(<ChatView messages={messages} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly and without crashing with a single message from the bot", () => {
  let messages = [new Message("Hi there", "bot")];

  const tree = renderer.create(<ChatView messages={messages} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly and without crashing with a single message from the bot", () => {
  let messages = [new Message("Hi there", "user")];

  const tree = renderer.create(<ChatView messages={messages} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly and without crashing with a multiple messages from the bot", () => {
  let messages = [
    new Message("Hi there", "bot"),
    new Message("How's it going", "bot")
  ];

  const tree = renderer.create(<ChatView messages={messages} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly and without crashing with a multiple messages from the user", () => {
  let messages = [
    new Message("Hi there", "user"),
    new Message("How's it going", "user")
  ];

  const tree = renderer.create(<ChatView messages={messages} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly and without crashing with mixed messages", () => {
  let messages = [
    new Message("Hi there", "user"),
    new Message("How's it going", "bot")
  ];

  const tree = renderer.create(<ChatView messages={messages} />).toJSON();
  expect(tree).toMatchSnapshot();
});
