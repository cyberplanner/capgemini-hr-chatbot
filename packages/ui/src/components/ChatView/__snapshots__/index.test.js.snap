// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and without crashing with a multiple messages from the bot 1`] = `
<div
  className="sc-bdVaJa eYYgnR"
>
  <p
    className="sc-bwzfXH biAvUV"
  >
    bot
  </p>
  <p
    className="sc-bwzfXH biAvUV"
  >
    bot
  </p>
</div>
`;

exports[`renders correctly and without crashing with a multiple messages from the user 1`] = `
<div
  className="sc-bdVaJa eYYgnR"
>
  <p
    className="sc-bwzfXH biAvUV"
  >
    user
  </p>
  <p
    className="sc-bwzfXH biAvUV"
  >
    user
  </p>
</div>
`;

exports[`renders correctly and without crashing with a single message from the bot 1`] = `
<div
  className="sc-bdVaJa eYYgnR"
>
  <p
    className="sc-bwzfXH biAvUV"
  >
    bot
  </p>
</div>
`;

exports[`renders correctly and without crashing with a single message from the bot 2`] = `
<div
  className="sc-bdVaJa eYYgnR"
>
  <p
    className="sc-bwzfXH biAvUV"
  >
    user
  </p>
</div>
`;

exports[`renders correctly and without crashing with mixed messages 1`] = `
<div
  className="sc-bdVaJa eYYgnR"
>
  <p
    className="sc-bwzfXH biAvUV"
  >
    user
  </p>
  <p
    className="sc-bwzfXH biAvUV"
  >
    bot
  </p>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="sc-bdVaJa eYYgnR"
/>
`;

exports[`renders without crashing with an empty array of messages 1`] = `
<div
  className="sc-bdVaJa eYYgnR"
/>
`;
